src/main/java/openmodularturrets/entity/projectiles/BlazingClayProjectile.java

```diff
@@ -68,25 +68,44 @@ protected void onImpact(MovingObjectPosition movingobjectposition) {
 
             int damage = ConfigHandler.getIncendiary_turret().getDamage();
 
-            if (isAmped) {
-                if (movingobjectposition.entityHit instanceof EntityLivingBase) {
-                    EntityLivingBase elb = (EntityLivingBase) movingobjectposition.entityHit;
-                    damage += ((int) elb.getHealth() * (0.05 * amp_level));
-                }
+            if (isAmped && movingobjectposition.entityHit instanceof EntityLivingBase) {
+                EntityLivingBase elb = (EntityLivingBase) movingobjectposition.entityHit;
+                damage += ((int) elb.getHealth() * (0.05 * amp_level));
             }
 
             for (Entity mob : targets) {
+                boolean wasAlive = !mob.isDead;
+                float healthBefore = 0;
+
+                if (mob instanceof EntityLivingBase) {
+                    EntityLivingBase elb = (EntityLivingBase) mob;
+                    healthBefore = elb.getHealth();
+
+                    if (mob instanceof EntityPlayer) {
+                        if (canDamagePlayer((EntityPlayer) mob)) {
+                            elb.attackEntityFrom(new NormalDamageSource("bullet"), damage);
+                            elb.hurtResistantTime = 0;
+                            elb.setFire(5);
+                        }
+                    } else {
+                        elb.attackEntityFrom(new NormalDamageSource("bullet"), damage);
+                        elb.hurtResistantTime = 0;
+                        elb.setFire(5);
+                    }
```

Comment:
Suggestion for simplifying:
```suggestion
                    if (!(mob instanceof EntityPlayer) || canDamagePlayer((EntityPlayer) player)) {
                        elb.attackEntityFrom(new NormalDamageSource("bullet"), damage);
                        elb.hurtResistantTime = 0;
                        elb.setFire(5);
                    }
```